### PR TITLE
Update Layout Examples to ASDK 2.0

### PR DESCRIPTION
**Pull request by hannahmbanana**
Separate layout code from layout playground so that we can make an identical swift project.

### CODE REVIEW

File: examples/LayoutSpecExamples/Sample/LayoutExampleNodes.h [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+@interface LayoutExampleNode : ASDisplayNode
++ (NSString *)title;
++ (NSString *)descriptionTitle;
+@end
+
+@interface HeaderWithRightAndLeftItems : LayoutExampleNode
+@property (nonatomic, strong) ASTextNode *usernameNode;
+@property (nonatomic, strong) ASTextNode *postLocationNode;
+@property (nonatomic, strong) ASTextNode *postTimeNode;
```
Comment (maicki) [inline examples/LayoutSpecExamples/Sample/LayoutExampleNodes.h]: We should move all of them out of the header into the class extension. So direct assigning values to the outside the node is not possible. Change of the state of this node should only happen by providing a new model.

Same for all other nodes in this file.

File: examples/LayoutSpecExamples/Sample/LayoutExampleNodes.m [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+  
+  return self;
+}
+
+- (ASLayoutSpec *)layoutSpecThatFits:(ASSizeRange)constrainedSize
+{
+  // when the username / location text is too long, shrink the stack to fit onscreen rather than push content to the right, offscreen
+  ASStackLayoutSpec *nameLocationStack = [ASStackLayoutSpec verticalStackLayoutSpec];
+  nameLocationStack.style.flexShrink = YES;
+  nameLocationStack.style.flexGrow = YES;
```
Comment (maicki) [inline examples/LayoutSpecExamples/Sample/LayoutExampleNodes.m]: Use 1.0

File: examples/LayoutSpecExamples/Sample/LayoutExampleNodes.m [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+    _titleNode.truncationAttributedText = [NSAttributedString attributedStringWithString:@"..." fontSize:16 color:[UIColor whiteColor]];
+    _titleNode.attributedText = [NSAttributedString attributedStringWithString:@"family fall hikes" fontSize:16 color:[UIColor whiteColor]];
+  }
+  
+  return self;
+}
+
+- (ASLayoutSpec *)layoutSpecThatFits:(ASSizeRange)constrainedSize
+{
+  _photoNode.style.preferredSize = CGSizeMake(constrainedSize.max.width / 4.0, constrainedSize.max.width / 4.0);
```
Comment (maicki) [inline examples/LayoutSpecExamples/Sample/LayoutExampleNodes.m]: As it seems like width and height has the same calculation, should we move that to a temporary variable above?

File: examples/LayoutSpecExamples/Sample/LayoutExampleNodes.m [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+
+  // INFINITY is used to make the inset unbounded
+  UIEdgeInsets insets = UIEdgeInsetsMake(INFINITY, 12, 12, 12);
+  ASInsetLayoutSpec *textInsetSpec = [ASInsetLayoutSpec insetLayoutSpecWithInsets:insets child:_titleNode];
+
+  ASOverlayLayoutSpec *textOverlaySpec = [ASOverlayLayoutSpec overlayLayoutSpecWithChild:_photoNode
```
Comment (maicki) [inline examples/LayoutSpecExamples/Sample/LayoutExampleNodes.m]: Return the overlay spec directly here.

File: examples/LayoutSpecExamples/Sample/LayoutExampleNodes.m [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+    _bottomSeparator.image = [UIImage as_resizableRoundedImageWithCornerRadius:1.0 cornerColor:[UIColor blackColor] fillColor:[UIColor blackColor]];
+  }
+  
+  return self;
+}
+
+- (ASLayoutSpec *)layoutSpecThatFits:(ASSizeRange)constrainedSize
+{
+  _topSeparator.style.flexGrow = YES;
+  _bottomSeparator.style.flexGrow = YES;
```
Comment (maicki) [inline examples/LayoutSpecExamples/Sample/LayoutExampleNodes.m]: Use 1.0

File: examples/LayoutSpecExamples/Sample/Utilities.m [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+@implementation NSAttributedString (Additions)
+
++ (NSAttributedString *)attributedStringWithString:(NSString *)string fontSize:(CGFloat)size color:(nullable UIColor *)color
+{
+  if (string == nil) {
+    return nil;
+  }
+  
+  NSDictionary *attributes                    = @{NSForegroundColorAttributeName: color ? : [UIColor blackColor],
```
Comment (maicki) [inline examples/LayoutSpecExamples/Sample/Utilities.m]: I think that's a weird alignment. Just move it to the `attributes` value without this space.

**Comment by maicki** [comment]
@hannahmbanana Just added a couple of comments, but overall looks good!

**Comment by hannahmbanana** [comment]
Ready to merge.

**Pull request merged**